### PR TITLE
Update dependencies for wider use

### PR DESCRIPTION
**Pull request by fenekku**
## Description

- remove conflicting (with other modules) dependencies
- update code to account for Marshmallow 3
- update code to use mock from unittest.mock i.e. update to Python 3
- Execution of RFC 21: removal of examples app
- make travis work
- bump version

### CODE REVIEW

**Comment by fenekku** [comment]
image

that was... not cool

File: setup.py [hunk 1]
```diff
@@ -70,16 +67,10 @@
 ]
 
 install_requires = [
-    'Flask-BabelEx>=0.9.3',
-    'flask-oauthlib>=0.9.4',
+    'Flask-BabelEx>=0.9.4',
```
Comment (slint) [inline setup.py]: **Minor:** That could also be `invenio-i18n` in order to apply the "coordinator" packages practice

File: invenio_pidrelations/utils.py [hunk 2]
```diff
@@ -1,32 +1,29 @@
...
 
-from __future__ import absolute_import, print_function
-
 import six
 from flask import current_app
 from werkzeug.utils import import_string
 
 
-def obj_or_import_string(value, default=None):
+def obj_or_import_string(value):
```
Comment (slint) [inline invenio_pidrelations/utils.py]: **Minor:** Actually this utility exists in `invenio-base`, so adding this as a dependency in `setup.py` and importing from there would make this definition here unnecessary.

File: setup.py [hunk 3]
```diff
@@ -70,16 +67,10 @@
...
     'invenio-pidstore>=1.0.0',
-    'invenio-rest[cors]>=1.0.0',
-    'marshmallow>=2.15.2',
-    'requests-oauthlib>=1.1.0,<1.2.0',
-    'oauthlib<3.0.0,>=2.1.0',
-    'six>=1.11',
-    'urllib3<1.25,>=1.21.1',
-    'idna<2.8,>=2.5',
+    'marshmallow>=3.3.0,<4.0.0',
+    'six>=1.12',
```
Comment (slint) [inline setup.py]: If you add `invenio-base`  (as mentioned above) this one is unnecessary

**Review by slint — APPROVED** [review]
LGTM! Some minor dependency management comments, but nothing blocking

File: setup.py (hunk 1 above)
Comment (fenekku) [reply setup.py]: will try it out!

File: invenio_pidrelations/utils.py (hunk 2 above)
Comment (fenekku) [reply invenio_pidrelations/utils.py]: Right. This is basically re-implemented in all invenio modules. Using the one defined in invenio-base sounds good :+1:

**Pull request merged**